cli/wallet/wallet.go

```diff
@@ -400,22 +426,24 @@ func convertWallet(ctx *cli.Context) error {
 }
 
 func addAccount(ctx *cli.Context) error {
-	wall, err := openWallet(ctx.String("wallet"))
+	wall, _, err := openWallet(ctx.String("wallet"), ctx.String("wallet-config"))
 	if err != nil {
 		return cli.NewExitError(err, 1)
 	}
 
 	defer wall.Close()
 
-	if err := createAccount(wall); err != nil {
+	// Do not use the provided password from wallet config file, otherwise it can be misleading
+	// (wallet may already have an account with the specified password).
+	if err := createAccount(wall, nil); err != nil {
```

Comment:
I think with `wallet-config` option this is not misleading at all. Actually, it is a feature, e.g. `change-password` command assumes all accounts have the same password.

Reply:
Yeah, reusing the password from the config is OK here.

Reply:
Fixed.